### ISSUE TITLE
[MT-5066] Calling Lists.unwrapList() produces a side effect of list with empty list-items

### ISSUE THREAD

**Issue by obeckman** [prezly/slate#207]
hello i am using 
```
"slate": "^0.70.0",
"slate-react": "^0.70.0",
"react": "17.0.1",
```

and wanted to use  `"@prezly/slate-lists": "^0.40.3"` (latest )
all went well until i wanted to toggle current selected nodes using `List.unwrapList()`:

- **on a single selected line** all is ok 

- **on multiple selected lines** - when we unwrap the selected list items are unwrapped but additional new empty items are created 
(i seen that Schema.createDefaultTextNode is invoked ) 

**_### the live demo is broken (for over a month) so i cant test this there .... (or to see proper working code )_**

my code :

```ts
if (sameAsCurrentListType) {
   //the bug
   Lists.unwrapList(editor) 
   //Lists.decreaseDepth(editor) (same as unwrapList)
} else {
   //switching between types of lists , this 2 lines of code works perfect 
   Lists.wrapInList(editor, listType)
   Lists.setListType(editor, listType)
}
```  
 
 ![Screen Shot 2022-06-07 at 15 40 22](https://user-images.githubusercontent.com/23051318/172381480-33074fb7-3d6b-46eb-84c9-61ee131a6eb9.png)    
![Screen Shot 2022-06-07 at 15 40 44](https://user-images.githubusercontent.com/23051318/172381471-d8577457-2e24-4bd7-9aec-825f901f1715.png)

**Comment by obeckman** [comment]
i have created a semy working sandbox you can play around with (i hope ) where you can see the code and the bug reproduce ... 
[codesandbox](https://codesandbox.io/s/prezly-slate-lists-demo-complete-example-forked-4yrukm?file=/src/Toolbar.tsx)

senario :

- select all text 
- mark as list 
- mark as paragraph 
notice the new lines added below 

be a ware this code version has many other bugs ....

**Comment by e1himself** [comment]
Thank you for the report :+1: 
I will take a look at it.

**Comment by obeckman** [comment]
additional related bug ...  on the unwrapList we get infinit loop (it might be the same bug )
create a list with  3 numbered bullets, select all of them ,  convert into paragraph .... boom

**Comment by obeckman** [comment]
any updates on this subject?

**Comment by obeckman** [comment]
i am pretty sure its the fix/ or partial  of it 
@prezly/slate-lists/src/lib/decreaseListItemDepth.ts
instead of :
nextSiblings.forEach(() => {
        // The next sibling path is always the same, because once we move out the next sibling,
        // another one will take its place.
        const nextSiblingPath = [...parentListPath, listItemIndex + 1];
        increaseListItemDepth(editor, nextSiblingPath);
    });
    
the fix :
    nextSiblings.forEach(() => {
        // The next sibling path is always the same, because once we move out the next sibling,
        // another one will take its place.
        const nextSiblingPath = [...parentListPath, listItemIndex + 1];
       if (Node.has(editor, nextSiblingPath)) {
            decreaseListItemDepth(editor, nextSiblingPath);
        }
    });

**Comment by e1himself** [comment]
@obeckman 

Sorry, I didn't have time to check it yet. Was busy with lots of other time. Will do my best to spend time on this this week. 

Apologies for the delay. And thank you for the suggested code. It'll definitely help :+1:

**Comment by e1himself** [comment]
Hi @obeckman :wave: 

I sincerely apologise for the delay. There was plenty of other work I had to deal with in the meanwhile. 

I believe this problem is fixed with #286. 

Please upgrade to [v0.54.5](https://github.com/prezly/slate/releases/tag/v0.54.5) or higher, and let me know if it's good now.
Meanwhile, I'm closing this issue. I'm ready to reopen it if the problem still persists for you.

Thank you again for reporting this bug :+1:
